src/PrestaShopBundle/Form/Admin/Improve/Shipping/Preferences/PreferencesFormDataProvider.php

```diff
@@ -0,0 +1,139 @@
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ *  @author PrestaShop SA <contact@prestashop.com>
+ *  @copyright  2007-2018 PrestaShop SA
+ *  @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ *  International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Form\Admin\Improve\Shipping\Preferences;
+
+use PrestaShop\PrestaShop\Adapter\Carrier\CarrierOptionsConfiguration;
+use PrestaShop\PrestaShop\Adapter\Carrier\HandlingConfiguration;
+use PrestaShop\PrestaShop\Core\Form\FormDataProviderInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * Class is responsible of managing the data manipulated using forms
+ * in "Improve > Shipping > Preferences" page.
+ */
+class PreferencesFormDataProvider implements FormDataProviderInterface
+{
+    /**
+     * @var HandlingConfiguration
+     */
+    private $handlingConfiguration;
+
+    /**
+     * @var CarrierOptionsConfiguration
+     */
+    private $carrierOptionsConfiguration;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    public function __construct(
+        HandlingConfiguration $handlingConfiguration,
+        CarrierOptionsConfiguration $carrierOptionsConfiguration,
+        TranslatorInterface $translator
+    ) {
+        $this->handlingConfiguration = $handlingConfiguration;
+        $this->carrierOptionsConfiguration = $carrierOptionsConfiguration;
+        $this->translator = $translator;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getData()
+    {
+        return [
+            'handling' => $this->handlingConfiguration->getConfiguration(),
+            'carrier_options' => $this->carrierOptionsConfiguration->getConfiguration(),
+        ];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setData(array $data)
+    {
+        /**
+         * All three following fields (shipping_handling_charges, free_shipping_price, free_shipping_weight)
+         * can be submitted as either empty or with "0" values
+         * and in both cases it means that the functionality is disabled.
+         * For consistency changing the values to 0, in case they were submitted empty.
+         */
+        $data['handling']['shipping_handling_charges'] = $data['handling']['shipping_handling_charges'] ?: 0;
```

Comment:
What you want to do here, is to declare a default value if empty: you can do it in your form ;)

```php
$builder
    ->add('shipping_handling_charges', MoneyWithSuffixType::class, [
            'currency' => $defaultCurrency->iso_code,
            'suffix' => $this->trans('(tax excl.)', 'Admin.Global'),
            'required' => false,
            'empty_data' => 0,
        ]);
```